hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/SevenToEightUpgradeHandler.java

```diff
@@ -62,19 +78,38 @@ public class SevenToEightUpgradeHandler implements UpgradeHandler {
   @Override
   public Map<ConfigProperty, String> upgrade(HoodieWriteConfig config, HoodieEngineContext context,
                                              String instantTime, SupportsUpgradeDowngrade upgradeDowngradeHelper) {
+    Map<ConfigProperty, String> tablePropsToAdd = new HashMap<>();
     HoodieTable table = upgradeDowngradeHelper.getTable(config, context);
     HoodieTableMetaClient metaClient = table.getMetaClient();
     HoodieTableConfig tableConfig = metaClient.getTableConfig();
+    // If auto upgrade is disabled, set initial version and writer version to 6 and return
+    if (!config.autoUpgrade()) {
+      setInitialVersion(config, table.getMetaClient().getTableConfig(), tablePropsToAdd);
+      config.setValue(HoodieWriteConfig.WRITE_TABLE_VERSION, String.valueOf(HoodieTableVersion.SIX.versionCode()));
+      return tablePropsToAdd;
+    }
```

Comment:
this block is to ensure the migration protocol i.e. users can upgrade the binaries but still continue to write in tv=6 (no auto upgrade).